### ISSUE TITLE
Setting window.opener to null doesn't work in Electron

### ISSUE THREAD

**Issue by codebykat** [microsoft/monaco-editor#2220]
TL;DR: I just resorted to spoofing the User-Agent in Electron to get `window.open` working with external windows. 🤦‍♀️ 

Similar issue to: https://github.com/microsoft/monaco-editor/issues/628

Electron uses [its own version](https://www.electronjs.org/docs/api/window-open) of `window.open`, which always returns `null` (does not return a handle to the window). There's now an option (`nativeWindowOpen`) to use the native `window.open`, which gets external links to work (opening a new Electron window), but if you need to open external links in the default browser instead, there's no way to get the URL Monaco is attempting to open. The initial `window.open()` request will trigger `new-window`, but without the `url` in that call, it's a dead end.

The upshot is, if you're using the common pattern of hooking to the `new-window` event to trap external links and open them in the default browser, "Follow Link" will silently fail (or, on Windows, open a dialog saying you have to choose an app to open "this about link", presumably because it doesn't have a default handler registered for "about:blank").

I unfortunately found this extremely hard to work around, because there's just no way to get the requested URL if the Opener receives a `null` in response to `window.open`. At one point I resorted to adding a `click` event to the editor DOM and checking to see if the click took place on a `detected-link-active`. 😰 I also tried the hack of allowing the window to open and listening for `will-navigate` events on it, then closing it and going to the requested URL; but since there's no way to modify the options on the initial `window.open`, it can't be loaded as a hidden window, so the user would see it open and immediately close before opening the browser window.

Adding "Electron" to the user-agent / platform detection and another boolean [here](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/dom.ts#L1214) would fix the problem.

Alternatively, maybe we could use some kind of option to toggle this behavior back to the plain `window.open(url)`.

My terrible hacks and eventual user-agent "fix" are here: https://github.com/Automattic/simplenote-electron/pull/2470

n.b. there are apparently Electron changes in the pipeline that will make this easier to work around, but it would still be good not to have to hack around it. https://github.com/electron/electron/pull/24517

**monaco-editor version:** 0.20.0
**Browser:** Electron 9.1.0 (tested in 10 and 11-beta as well)
**OS:** OSX Catalina, also reproduced on Windows 10
**Playground code that reproduces the issue:**

( Any that contains a detectable link )

Electron external window setup:
```
    mainWindow.webContents.on('new-window', function (event, linkUrl) {
      event.preventDefault();
      shell.openExternal(linkUrl);
    });
```

**Comment by codebykat** [comment]
> alexdima closed this in microsoft/vscode@d746f2d 8 days ago

@alexdima and @nrayburn-tech Thank you for the fix! 👏
